### ISSUE TITLE
[V3][field lifecycle] remove onChanges

### ISSUE THREAD

**Issue by aitboudad** [formly-js/ngx-formly#643]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[x] feature request
[ ] support request
```

since `onChanges` doesn't work properly with dynamic component and also it appear a little confusing (#397, #390) I propose to remove it.

**Comment by BirgitPohl** [comment]
Do you have any suggestion what workarounds to use instead?

**Comment by aitboudad** [comment]
nop

**Comment by BirgitPohl** [comment]
Then let me give the folks one.
Adding a standard event listener is one solution.

```
ngAfterViewInit() {
    setTimeout(() => {
        element.addEventListener('change', function(event) {
          doSomething{();
        })
      }
    }, 1000);
  }
```

**Comment by aitboudad** [comment]
Note: `onChanges` is not about tracking change of model but component inputs changes.

**Comment by diegochavez** [comment]
This issue cause a lag interacting with elements inside of formly fields? I have a strange behaviour related to this but I'm searching on the issues if theres something to improve with on change detection.

**Comment by aitboudad** [comment]
@diegochavez can you elaborate more on your strange behavior ? providing a reproduction example would be nice. Thanks.

**Comment by diegochavez** [comment]
Hello @aitboudad thanks man yeah I'll create an issue to explain better my case maybe is not related to this. but I have dynamically generated forms from REST API, but the behaviour inside of the forms is weird because after a couple of minutes of interaction the inputs are really slow to respond.

**Comment by aitboudad** [comment]
got a way to make it work, it should be available in `v3.0`.
